environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java

```diff
@@ -113,6 +118,28 @@ public ValidationResult validateSecurityGroups(EnvironmentEditDto editDto, Envir
 
     }
 
+    public ValidationResult validateAuthenticationModification(EnvironmentEditDto editDto, Environment environment) {
+        ValidationResult.ValidationResultBuilder validationResultBuilder = ValidationResult.builder();
+
+        AuthenticationDto authenticationDto = editDto.getAuthentication();
+        String publicKeyId = authenticationDto.getPublicKeyId();
+        if (!environment.getCloudPlatform().equals(CloudPlatform.AWS.name()) && StringUtils.isNotEmpty(publicKeyId)) {
```

Comment:
can we somehow do this more dynamic so the the cloudconnector giving us this info ?